### PR TITLE
Fix conflict between `RtcClock::get_xtal_freq` and `Rtc::disable_rom_message_printing`

### PR DESCRIPTION
**Pull request by dimpolo**
First commit is just some deduplication and cleanup.

See also:
https://github.com/espressif/esp-idf/commit/c7c2462d39dff90a9c2f0ab1ef8fe3c4a1ecea86

Tested on the ESP32S3

### CODE REVIEW

File: esp-hal/src/rtc_cntl/mod.rs [hunk 1]
```diff
@@ -423,7 +425,7 @@ impl<'d> Rtc<'d> {
         let rtc_cntl = unsafe { lp_aon() };
         rtc_cntl
             .store4()
-            .modify(|r, w| unsafe { w.bits(r.bits() | 1) });
+            .modify(|r, w| unsafe { w.bits(r.bits() | Self::RTC_DISABLE_ROM_LOG) });
```
Comment (bugadani) [inline esp-hal/src/rtc_cntl/mod.rs:428]: We might be better off adding this bit to the PAC, and that would also allow us to remove the masking of this bit in L508.

Comment (dimpolo) [reply esp-hal/src/rtc_cntl/mod.rs:428]: My understanding is that these registers don't have actual functionality associated with them. Instead they're a way to pass data between the bootloaders and the application. Not sure if that's a PAC thing

Comment (bugadani) [reply esp-hal/src/rtc_cntl/mod.rs:428]: I guess that's technically correct, but if we use this as something fixed-function, we might as well make our lives a bit easier :) Not sure what the team thinks, though, it's just my opinion.

**Review by bugadani — APPROVED** [review]
Thanks, let's get this in before it starts collecting dust and merge conflicts.

File: esp-hal/src/rtc_cntl/mod.rs (hunk 1 above)
Comment (bjoernQ) [reply esp-hal/src/rtc_cntl/mod.rs:428]: I'd say it's just a register to store arbitrary data so probably it's better to keep it as is. `esp-hal` might not be the only consumer of the SVDs

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**